### ISSUE TITLE
Inconsistent handling of JSON

### ISSUE THREAD

**Issue by garrytrinder** [PowerShell/PowerShell#11757]
Raised from https://github.com/pnp/office365-cli/issues/1266

# Steps to reproduce

On MacOS, execute a command that returns a large JSON output and convert output from JSON using ConvertFrom-Json cmdlet, e.g. `o365 spo site classic list -o json | ConvertFrom-Json`

# Expected behavior

JSON should be converted to PSObject

# Actual behavior

JSON is not converted and error is thrown

```none
ConvertFrom-Json : Conversion from JSON failed with error: Unexpected end of content while loading JArray. Path '[28].RelatedGroupId', line 1, position 65534.
+ o365 spo site classic list -o json | ConvertFrom-Json
+                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [ConvertFrom-Json], ArgumentException
+ FullyQualifiedErrorId : System.ArgumentException,Microsoft.PowerShell.Commands.ConvertFromJsonCommand
```

`o365 spo site classic list -o json` returns the full JSON output to the terminal as expected, however when saving this output to a variable e.g. `$json = o365 spo site classic list -o json` the output is trimmed.

Redirecting the output to a file e.g. `o365 spo site classic list -o json > pwshsites.json` saves a file with trimmed output.

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      6.2.4
PSEdition                      Core
GitCommitId                    6.2.4
OS                             Darwin 19.2.0 Darwin Kernel Version 19.2.0: Sat Nov  9 03:47:04 PST 2019; root:xnu-6153.61.…
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

# Additional information

The `o365` command used in the example repro steps refers to the Office 365 CLI which I am one of the project maintainers.

The referenced issue has further investigation, however this behaviour is only seen on MacOS, having tested on Windows PowerShell and PowerShell Core on Windows, the JSON output is parsed successfully.

**Comment by iSazonov** [comment]
ConvertFrom-Json is based on NewtonSoft.Json and the issue looks like an issue in the framework.

Current plan is to migrate to .Net Core API. The work already started in #11198 for ConvertTo-Json. I do not know when it will be available in public (v7.1?).

You could open an issue in NewtonSoft.Json repo and we would get it with servicing release (v7.0.1?).

**Comment by garrytrinder** [comment]
Thank you for the response @iSazonov and for the explanation, no problem in raising an issue 👍🏻
